From c532b76ab9210e58993f563d8e48140aa0b441cb Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Sep 2024 10:32:45 +0300
Subject: [PATCH] [FSW-6580] fs1: set lfs done after boot info get

Boot rom has to read and verify boot info data
from lfs files in non cold reset scenarios. Once
data get completes boot rom will set lfs done
progress bit.

Change-Id: I5ea68abff70720f131d827520b895d8a77a6649d
---

diff --git a/arch/fs1/arc/dev_boot_info.c b/arch/fs1/arc/dev_boot_info.c
index 56ea156..27770a6 100644
--- a/arch/fs1/arc/dev_boot_info.c
+++ b/arch/fs1/arc/dev_boot_info.c
@@ -52,3 +52,18 @@
 
 	return;
 }
+
+void boot_rom_get_boot_info(void)
+{
+	if (boot_rom_global.boot_info_set_done) {
+		/* Read and make sure proper data after any reset */
+		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START);
+
+		/* TODO: retrieve boot info and verify */
+
+		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH);
+	}
+
+	/* Flag which is indication for other modules to know Boot ROM done with LFS */
+	boot_rom_status_update(PROGRESS_BOOT_INFO_LFS_FINISHED);
+}
diff --git a/arch/fs1/arc/dev_boot_info.h b/arch/fs1/arc/dev_boot_info.h
index 52da37a..186feed 100644
--- a/arch/fs1/arc/dev_boot_info.h
+++ b/arch/fs1/arc/dev_boot_info.h
@@ -14,5 +14,6 @@
 #define CONFIG_BOOTINFO_SIZE	OFFSET_32KB
 
 void boot_rom_set_boot_info(void);
+void boot_rom_get_boot_info(void);
 
 #endif	/* _DEV_BOOT_INFO_H_ */
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 7f2e564..a9b02d9 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -17,6 +17,7 @@
 #include <dma.h>
 #include <common.h>
 #include <boot_flow.h>
+#include <dev_boot_info.h>
 
 uint32_t fake_fuse_offset_array[MAX_DIE_COUNT];
 
@@ -176,6 +177,8 @@
 skip_boot_desc_process:
 	p_manifest_ctx->boot_desc_flag = 1;
 
+	boot_rom_get_boot_info();
+
 	return BOOT_ROM_SUCCESS;
 }
 
